health/checks.go

```diff
@@ -40,21 +40,33 @@ func TCPDialCheck(addr string, timeout time.Duration) Check {
 // specified URL. The check fails if the response times out or returns a non-200
 // status code.
 func HTTPGetCheck(url string, timeout time.Duration) Check {
+	return func() error {
+		return HTTPCheck(url, http.MethodGet, http.StatusOK, timeout)()
+	}
+}
+
+// HTTPCheck returns a Check that performs a HTTP request against the specified URL.
+// The Check fails if the response times out or returns an unexpected status code.
+func HTTPCheck(url string, method string, status int, timeout time.Duration) Check {
 	client := http.Client{
 		Timeout: timeout,
-		// never follow redirects
 		CheckRedirect: func(*http.Request, []*http.Request) error {
 			return http.ErrUseLastResponse
 		},
 	}
+
 	return func() error {
-		resp, err := client.Get(url)
+		req, err := http.NewRequest(method, url, nil)
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Do(req)
 		if err != nil {
 			return err
 		}
-		resp.Body.Close()
-		if resp.StatusCode != 200 {
-			return fmt.Errorf("returned status %d", resp.StatusCode)
+		if resp.StatusCode != status {
```

Comment:
The response body should still need to be closed.

```suggestion
		resp.Body.Close()
		if resp.StatusCode != status {
```

Reply:
We don't read it at all? Are you sure? 🤔

Reply:
I think so.`If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close.`
https://golang.org/pkg/net/http/#Client.Do

It's a good practice to make sure all underlying resources released.

Reply:
Ok. Good. I am aware but thought that when not reading the body, it's also fine to not defer the close. I will add it to better be safe than sorry. :)